### PR TITLE
[FIX] base_exception: Migration scripts

### PR DESCRIPTION
**Pull request by pedrobaeza**
Current method doesn't work, as this module is not being migrated, but fresh-installed by dependencies of old `sale_exception`, so we should use the pre_init_hook and detect if we should run the migration code according the existence of the table.

cc @Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@mourad-ehm please review, as you were the original author.

**Comment by pedrobaeza** [comment]
@bealdav maybe you can review this one.

**Review by bealdav — APPROVED** [review]
of course

File: base_exception/hooks.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+def pre_init_hook(cr):
+    """If coming from v9 where only `sale_exception` is installed, when
+    auto-installing this one by dependencies, we should take care of
+    renaming previous table before the new model is loaded.
+    """
+    cr.execute("SELECT 1 FROM pg_class WHERE relname = 'sale_exception'")
+    if not cr.fetchone():
+        return  # fresh install, so exiting
+    from openupgradelib import openupgrade
```
Comment (mourad-ehm) [inline base_exception/hooks.py:15]: As we are not in the context of openupgrade,  penupgradelib is an external lib. Should not be nice to add and external dependency in manifest ? And manage import exception like: 
try:
   from openupgradelib import openupgrade
except ImportError:
    ....
?

Comment (pedrobaeza) [reply base_exception/hooks.py:15]: Not needed. That's why I import after the checking of the column: if you are fresh installing the module, no need of this lib. But if you come from previous version, then you have already openupgradelib due to OpenUpgrade, so also not a problem.

Comment (mourad-ehm) [reply base_exception/hooks.py:15]: Ok

**Review by mourad-ehm — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Thanks for reviewing

**Pull request merged**